modules/nodepool/main.tf

```diff
@@ -29,8 +29,10 @@ resource "aws_launch_template" "this" {
     security_groups             = concat(var.vpc_security_group_ids, [aws_security_group.this.id])
   }
 
+#FIXME: maybe device should be a variable? 
+#    device_name = lookup(var.block_device_mappings, "device_name", "/dev/sda1")
   block_device_mappings {
-    device_name = lookup(var.block_device_mappings, "device_name", "/dev/sda1")
+    device_name = lookup(var.block_device_mappings, "device_name", "/dev/xvda")
```

Comment:
Why are we changing the mount location?

Reply:
apparently it differs per AMI being used, which isn't apparent by just looking at this PR. 

We should probably investigate using the [terraform `aws_ami` data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami) to query the AMI and then figure out what the device name is supposed to be. 

One of the output attributes is the `block_device_mappings` which has an attribute on each element for the `device_name`

Reply:
Huh, Today I learned. I could've sworn `sda` was specifically for root and/or block devices and that `xvd` was for extensions. I feel like the most recent time I saw that being used was for NVME SSDs.

Good to know.

Reply:
This gave us fits for a few days when the gp2/gp3 rules changed .  I think it depends on the bus type - for example /dev/sd is a scsi bus, /dev/hd is ide, xvd is xen paravirtual.  The next character is the order of the device on the bus ( a is the first device, b the second ...) and the trailing number is the partition number ( I think ).  Additionally , I think its possible that the drivers loaded might change the device names.  If you have the xen driver loaded - you should see the xvd device names, if not and you are using stock scsi - then you will see the sda.

I like the terraform aws_ami data source idea and think there are likely other places we might be able to do something similar to reduce the amount of stuff users have to provide - for example if I know the domain name I want to use - I can look up the hosted zone id .